Type issue: SerializableValueType is too restrictive when it comes to sql.json

<!--- Provide a general summary of the issue in the Title above -->
Just upgraded to latest version of slonik, and many of my calls to sql.json are now reporting type errors of `Type 'MyCustomType' is not assignable to type '{ [key: string]: SerializableValueType; }'`. The issue is that if MyCustomType is an object type that contains any optional fields, then the TS type of those fields will include undefined, which SerializableValueType

## Expected Behavior
<!--- Tell us what should happen -->
If I have a type defined as something like `type MyCustomType = { foo: string; bar?: string };` then a should be able to use an instance of that type in a call to sql.json

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
Error message as described above

## Possible Solution
<!--- (optional) Suggest a fix/reason for the bug, -->
Type definition for SerializableValueType should be the following I believe (see [key: string]: SerializableValueType **| undefined**):
```
export declare type SerializableValueType = string | number | boolean | null | {
    [key: string]: SerializableValueType | undefined;
} | ReadonlyArray<SerializableValueType>;
```

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

1. See comments above

## Logs

<!-- Capture and include logs https://github.com/gajus/slonik#logging -->

```
error TS2345: Argument of type 'MyCustomType' is not assignable to parameter of type 'SerializableValueType'.
  Type 'MyCustomType' is not assignable to type '{ [key: string]: SerializableValueType; }'.
    Property 'm' is incompatible with index signature.
      Type 'string | undefined' is not assignable to type 'SerializableValueType'.
        Type 'undefined' is not assignable to type 'SerializableValueType'.
```

--- a/.eslintrc
+++ b/.eslintrc
@@ -14,6 +14,7 @@
     "@typescript-eslint/no-non-null-assertion": 1,
     "@typescript-eslint/prefer-ts-expect-error": 1,
     "unicorn/prevent-abbreviations": 0,
+    "unicorn/numeric-separators-style": 0,
     "no-undef": 0,
     "linebreak-style": 0,
     // https://github.com/typescript-eslint/typescript-eslint/issues/1624#issuecomment-589731039

--- a/src/types.ts
+++ b/src/types.ts
@@ -39,7 +39,7 @@ export type SerializableValueType =
   | boolean
   | null
   | {
-      [key: string]: SerializableValueType;
+      [key: string]: SerializableValueType | undefined;
     }
   | ReadonlyArray<SerializableValueType>;
 

--- a/test/slonik/templateTags/sql/json.ts
+++ b/test/slonik/templateTags/sql/json.ts
@@ -122,3 +122,18 @@ test('the resulting object is immutable', (t) => {
     token.foo = 'bar';
   });
 });
+
+test('Object types with optional properties are allowed', (t) => {
+  type TypeWithOptionalProperty = { foo: string; opt?: string };
+  const testValue: TypeWithOptionalProperty = {foo: 'bar'};
+
+  const query = sql`SELECT ${sql.json(testValue)}`;
+
+  t.deepEqual(query, {
+    sql: 'SELECT $1',
+    type: SqlToken,
+    values: [
+      '{"foo":"bar"}',
+    ],
+  });
+});

